operations_on_datastructures/trie_multiple_search.cpp

```diff
@@ -0,0 +1,470 @@
+/**
+ * @file
+ * @brief [Trie datastructure](https://iq.opengenus.org/autocomplete-using-trie-data-structure/)
+ * with search variants
+ * @details
+ * This provides multiple variants of search functions
+ * on a trie structure utilizing STL. The trie is valid
+ * for only English alphabets.
+ * @author [Ghanashyam](https://github.com/g-s-k-zoro)
+ */
+
+#include <cassert>   /// for assert
+#include <cctype>    /// for tolower
+#include <cstring>   /// for string operations
+#include <iostream>  /// for IO Operations
+#include <queue>     /// for std::priority_queue
+
+/**
+ * @namespace operations_on_datastructures
+ * @brief Operations on data structures
+ */
+namespace operations_on_datastructures {
+/**
+ * @namespace trie_operations
+ * @brief Functions for [Trie datastructure](https://iq.opengenus.org/autocomplete-using-trie-data-structure/)
+ * implementation
+ */
+namespace trie_operations {
+/**
+ * @brief Class defining the structure of trie node and containing the methods
+ * to perform operations on them.
+ */
+class Tnode {
+ private:
+    static constexpr uint8_t ENGLISH_ALPHABET_SIZE = 26;
+    // pointers to alphabets
+    std::vector<Tnode *> english;
+
+    // To mark the end of word
+    bool endOfWord;
+
+    // To store the frequency of searches for the word
+    uint32_t frequency;
+
+ public:
+    Tnode() {
+        english.resize(ENGLISH_ALPHABET_SIZE, nullptr);
+        endOfWord = false;
+        frequency = 0;
+    }
+    // Copy Constructor
+    Tnode(const Tnode &node) {
+        english = node.english;
+        endOfWord = node.endOfWord;
+        frequency = node.frequency;
+    }
+
+    Tnode &operator=(const Tnode &node) = default;
+
+    Tnode(Tnode &&) = default;
+
+    Tnode &operator=(Tnode &&) = default;
+    /**
+     * @brief Function to count the number of children a node in the trie has
+     * @param node a trie node whose children need to be counted
+     * @return count of the number of children of the given node (max 26)
+     */
+    inline uint8_t numberOfChildren(Tnode *node) {
+        uint8_t count = 0;
+        for (auto &i : node->english) {
+            if (i) {
+                count++;
+            }
+        }
+        return count;

```

Comment:
A better way could be
```C++
return count = ENGLISH_ALPHABET_SIZE - std::count(node->english.begin(), node->english.end(), nullptr);
```

Reply:
Hey Ayaan, I tried that, but the lint recognizes it as one of the warning-as-errors. I can circumvent it through idiom but was thinking of keeping the original code. Want to avoid including\<algorithm\>header just for count. Is it ok if i proceed with the original way?

Reply:
we can proceed in any way, there was no problem with that both codes have the same asymptotic complexity and almost same runtime. Moreover `queue` internally uses `algorithms`, but let's leave this as it is.

Reply:
> we can proceed in any way, there was no problem with that both codes have the same asymptotic complexity and almost same runtime. Moreover `queue` internally uses `algorithms`, but let's leave this as it is.

oh ok. Then i think using std::count is better. Have implemented it and pushed for your final review.